src/Lykke.Job.TransactionHandler/Handlers/EthereumCoreCommandHandler.cs

```diff
@@ -0,0 +1,468 @@
+            }
+        }
+
+        public async Task<CommandHandlingResult> Handle(SaveEthInHistoryCommand command, IEventPublisher eventPublisher)
+        {
+            try
+            {
+                var cashinId = command.CashinOperationId.ToString("N");
+                var clientId = command.ClientId;
+                var hash = command.TransactionHash;
+                var amount = command.Amount;
+                var asset = await _assetsServiceWithCache.TryGetAssetAsync(command.AssetId);
+                var clientAddress = command.ClientAddress;
+
+                var walletCreds = await _walletCredentialsRepository.GetAsync(clientId);
+                await _cashOperationsRepositoryClient.RegisterAsync(new CashInOutOperation
+                {
+                    Id = cashinId,
+                    ClientId = clientId,
+                    Multisig = walletCreds.MultiSig,
+                    AssetId = asset.Id,
+                    Amount = (double)amount,
+                    BlockChainHash = hash,
+                    DateTime = DateTime.UtcNow,
+                    AddressTo = clientAddress,
+                    State = TransactionStates.SettledOnchain
+                });
+
+                var clientAcc = await _clientAccountClient.GetByIdAsync(clientId);
+                await _srvEmailsFacade.SendNoRefundDepositDoneMail(clientAcc.PartnerId, clientAcc.Email, amount, asset.Id);
+                await _paymentTransactionsRepository.SetStatus(hash, PaymentStatus.NotifyProcessed);
+
+                eventPublisher.PublishEvent(new EthCashinSavedInHistoryEvent()
+                {
+                    CashinOperationId = cashinId,
+                    TransactionHash = hash
+                });
+
+                return CommandHandlingResult.Ok();
+            }
+            catch (Exception e)
+            {
+                _log.WriteError(command.TransactionHash ?? "Empty", command, e);
+                throw;
+            }
+        }
+
+        #endregion
+
+        #region Private
+
+        private async Task<bool> ProcessHotWalletCashin(ProcessHotWalletEventCommand queueMessage,
+            IEventPublisher eventPublisher)
+        {
+            var bcnCreds = await _bcnClientCredentialsRepository.GetByAssetAddressAsync(queueMessage.FromAddress);
+            string tokenAddress = queueMessage.TokenAddress;
+            var token = await _assetsService.Erc20TokenGetByAddressAsync(tokenAddress);
+            var asset = await _assetsServiceWithCache.TryGetAssetAsync(token.AssetId);
+            var amount = EthServiceHelpers.ConvertFromContract(queueMessage.Amount, asset.MultiplierPower, asset.Accuracy);
+
+            await HandleCashInOperation(asset, amount, bcnCreds.ClientId,
+                bcnCreds.Address, queueMessage.TransactionHash, eventPublisher);
+
+            return true;
+        }
+
+        private async Task<bool> ProcessHotWalletCashout(ProcessHotWalletEventCommand queueMessage)
```

Comment:
Result is always true and it's not used by the caller - should be void

Reply:
fixed